assets/js/blocks/price-filter/block.js

```diff
@@ -59,14 +95,42 @@ const PriceFilterBlock = ( { attributes, isEditor = false } ) => {
 	// Updates the query based on slider values.
 	const onSubmit = useCallback(
 		( newMinPrice, newMaxPrice ) => {
-			setMinPriceQuery(
-				newMinPrice === minConstraint ? undefined : newMinPrice
-			);
-			setMaxPriceQuery(
-				newMaxPrice === maxConstraint ? undefined : newMaxPrice
-			);
+			if ( filteringForPhpTemplate && window ) {
+				// For block templates that render the PHP Classic Template block we need to add the filters as params and reload the page.
+				const url = new URL( window.location );
+				const currentParams = new URLSearchParams( url.search );
+				const newParams = formatParams( url, {
+					max_price: {
+						value: newMaxPrice,
+						constraint: maxConstraint,
+					},
+					min_price: {
+						value: newMinPrice,
+						constraint: minConstraint,
+					},
+				} );
+
+				// If the params have changed, lets reload the page.
+				if ( currentParams.toString() !== newParams ) {
+					window.location =
+						url.origin + url.pathname + '?' + newParams;
+				}
+			} else {
+				setMinPriceQuery(
+					newMinPrice === minConstraint ? undefined : newMinPrice
+				);
+				setMaxPriceQuery(
+					newMaxPrice === maxConstraint ? undefined : newMaxPrice
+				);
```

Comment:
As I mentioned above, I am not sure the whole `value/constraint` thing is required. Since the constraint is always equality, it seems, and the evaluation is done already on L120 and 123, why not do that anyways before passing it to the `formatParams` function, like we do it before passing it to the `setMinPriceQuery`?

Something like:

```js
// Using snake_case, just for ease of assignment to URL params, but we can jump
// an extra hoop and assign it later.
const max_price = newMaxPrice === maxConstraint ? undefined : newMaxPrice;
// ...

const newParams = formatParams( url, { max_price, min_price } );

// ...else
setMinPriceQuery(min_price);
setMaxPriceQuery(max_price);
```

Reply:
Nice observation, you're totally right.